### PR TITLE
Update vscode-languageclient and enable proposed features

### PR DESCRIPTION
**Pull request by bubba**

### CODE REVIEW

**Comment by bubba** [comment]
Forgot to mention that some of the dependencies were updated through `npm audit fix`

File: package.json [hunk 1]
```diff
@@ -234,11 +233,11 @@
...
     "cheerio": "^1.0.0-rc.2",
-    "js-yaml": "^3.11.0",
-    "lodash": "^4.17.10",
+    "js-yaml": "^3.13.1",
+    "lodash": "^4.17.11",
     "lru-cache": "^4.1.3",
     "request": "^2.86.0",
     "request-promise-native": "^1.0.5",
-    "vscode-languageclient": "^4.4.0"
+    "vscode-languageclient": "5.3.0-next.4"
```
Comment (alanz) [inline package.json]: This looks like an "under development" version?

Comment (bubba) [reply package.json]: Yes, I'm not sure when it is planned on being released. This PR is not blocking anything though so we can wait until there is a stable release

Comment (alanz) [reply package.json]: I would hate to break things for users out there.

**Comment by alanz** [comment]
@bubba I am pretty sure this should be fixable now, given vscode 1.35.0 is now out.

**Comment by alanz** [comment]
And according to https://www.npmjs.com/package/vscode-languageclient, current version is 5.2.1

**Comment by bubba** [comment]
@alanz conflicts resolved!

File: package.json [hunk 2]
```diff
@@ -245,7 +245,7 @@
     "lru-cache": "^4.1.3",
     "request": "^2.86.0",
     "request-promise-native": "^1.0.5",
-    "vscode-languageclient": "^4.4.0",
+    "vscode-languageclient": "5.3.0-next.5",
```
Comment (alanz) [inline package.json]: Doesn't the `next` suffix mean you have to be on the dev channel for vscode?

**Comment by alanz** [comment]
@bubba I think the vscode-languageclient needs to be 5.2.1

File: package.json (hunk 2 above)
Comment (bubba) [reply package.json]: No, the vscode version is constrained from the `engines.vscode` field from what I can tell: I tried bumping it to 1.35 today and I got an error when installing it with my older version of vscode. `vscode-languageclient` is just the library for connecting vscode extension APIs to LSP

**Comment by bubba** [comment]
The progress APIs/plugs are only in 5.3.x https://github.com/microsoft/vscode-languageserver-node/commit/dbe64946d7e41a50996a7c2929ac4aa4911b505d

**Comment by bubba** [comment]
@alanz is this good to go?

**Comment by alanz** [comment]
@bubba yes, will do a release today some time

**Pull request merged**